### PR TITLE
Merge release/dec-rollup

### LINKED ISSUE #8332: Update msal dependency
azure-identity requires msal<1.0.0. We should update this when msal 1.0.0 is available.

### LINKED ISSUE #8339: InteractiveBrowserCredential doesn't cache tokens
The following opens a login page twice, even when the first login is successful:
```py
from azure.identity import InteractiveBrowserCredential

credential = InteractiveBrowserCredential()

for _ in range(2):
    credential.get_token('https://vault.azure.net/.default')
```

### LINKED ISSUE #8340: InteractiveBrowserCredential fails on Python 2.7
Some class isn't new-style:
```sh
Python 2.7.15 (v2.7.15:ca079a3ea3, Apr 30 2018, 16:30:26) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from azure.identity import InteractiveBrowserCredential
>>> InteractiveBrowserCredential().get_token('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "azure\identity\_internal\exception_wrapper.py", line 23, in wrapper
    raise_from(auth_error, ex)
  File "C:\27\lib\site-packages\six.py", line 737, in raise_from
    raise value
azure.core.exceptions.ClientAuthenticationError: Authentication failed: super() argument 1 must be type, not classobj
```

### PR DESCRIPTION
**Pull request by chlowell**
Includes commits up to azure-identity 1.0.1.

Closes #8332, closes #8339, closes #8340

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by chlowell** [comment]
@weshaggard I think this is ready to merge

File: sdk/core/azure-core/HISTORY.md [hunk 1]
```diff
@@ -5,6 +5,8 @@
 ## 1.1.0b1 Unreleased
 
 
+## Version 1.0.1
```
Comment (weshaggard) [inline sdk/core/azure-core/HISTORY.md]: Where there no release notes for this version?

Comment (weshaggard) [reply sdk/core/azure-core/HISTORY.md]: nevermind I see it didn't have any fixes yet just had it's version bumped.

**Pull request merged**